exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/TypeInferenceUtils.java

```diff
@@ -394,31 +414,37 @@ public RelDataType inferReturnType(SqlOperatorBinding opBinding) {
   }
 
   private static class DrillConcatSqlReturnTypeInference implements SqlReturnTypeInference {
-    private static final DrillConcatSqlReturnTypeInference INSTANCE = new DrillConcatSqlReturnTypeInference();
```

Comment:
For the various SqlReturnTypeInference, seems the same logic is applied here and in DrillConcatFuncHolder.  If we choose not to expand "FunctionScope" and model it as ReturnTypeInference, then we probably will be able to abstract and share the common code. That would save effort in the future to maintain two almost identical pieces of codes.